### DISCUSSION TITLE
Linux : QjackCtl with web video setup ?

### DISCUSSION THREAD

**Discussion by pcar75** [jamulussoftware/jamulus#2115] [Technical Support & Help] [answered]
Bonjour, 
I'm trying to share my electric guitar and voice, and either a recorded tune (OK) or a web tune (not OK) in Jamulus session under Linux. Can anyone help ?

With QjackCtl and PulseAudio, I can simultaneously play my guitar-voice and a recorded tune respectively to fellow musicians.  However, I cannot do the same with my guitar-voice and a video-audio web tune; the web page (You Tube or other web videos) freezes, no sound , video seems stuck.  If I shut down Jamulus and QjackCtl, I can view and hear (in my headphones) web videos without problems.  If the answer is to set Firefox to use PulseAudio or Jack, I failed to see how it's done.

Since there are people using both video (e.g.: meeting, YouTube) web app and Jamulus, it must be feasible !?

My setup :
I boot Ubuntu 20.04LTS + QjackCtl 0.6.2 + Jamulus 3.6.2 + other good sw  from a USB3.0 key 
and my PC tower still has a lot of free CPU, RAM, SSD and good video card  after 6 years.

Guitar -> Line6 ME-70 multi-effets -> Behringer Q802USB mixer-> computer
Mic -> Behringer Q802USB mixer -> computer
Behringer Q802USB mixer -> headphones

Mixer -> Jack -> Jamulus in
VLC Player -> PulseAudio -> Jack -> Jamulus in
Jamulus out -> Mixer

I'm no Linux wiz, but thanks for any info.

**Comment by rdica** [comment]
I use a program called jack_mixer. In it I have created mixer channels for most inputs I use; pulse, mpd, software synths, soundcard hardware INs (I have 4). I connect the apps and hardware INs to their respective jack_mixer channel ports, then connect the MAIN output to jamulus INs.

This gives me individual volume control over every sound producing app, jack-aware and pulse-based. I can play music files, youtube videos, mic, guitar, separately or simultaneously and input it all into jamulus and/or wherever I choose to route.

![jack_mixer](https://user-images.githubusercontent.com/16609801/140680210-dede027d-299f-4266-ba76-f6967563e2f4.png)

![qjack-20211107](https://user-images.githubusercontent.com/16609801/140680043-1388274d-75c9-40fc-be6a-639ffd4c556a.png)

If this helps you, we can discuss further...

**Comment by ignotus666** [comment]
Go to 'Sound preferences' and in the Output tab select 'Jack sink (Pulseaudio JACK sink)'. That will route 'system' audio (browsers, VLC, etc) out through Pulseaudio.

Oh, and +1 for Jack Mixer - it's a great way to control the volume for multiple audio streams. I use it in combination with a MIDI controller and couldn't live without it.

**Comment by jujudusud** [comment]
VLC has a built in native JACK connections instead of using ALSA and PulseAudio. Check in the preference menu.

For all the web things, I use Chromium instead of Firefox because I can't have the sound from FF.

**Reply by ignotus666** [reply]
Yes, it does, but then whenever you want to use VLC without JACK you have to go into the preferences menu and switch it back (and vice versa), which is a pain in the behind.

**Reply by rdica** [reply]
Since there's always going to be a pulse sink, why not leave vlc configured to use it all the time. I understand there might be a use case for vlc using jack, but unless your use case is to broadcast from it all the time and you need jack sources for the broadcast, I suggest leaving vlc to use pulse.

**Reply by rdica** [reply]
@jujudusud I don't understand "I can't have the sound from FF."
Why can't you?
FF audio outputs to pulse for me. I can play videos in youtube etc just fine. All audio output from FF uses pulse sink.

**Reply by jujudusud** [reply]
You are right for Pulse sink + Firefox but, on my machine, it messes my connections between PulseAudio and JACK. Using Chromium solves this problem.

**Comment by pcar75** [comment, marked as answer]
Thanks for these leads,  I'm running tests and getting some positive results. I will publish my findings here.
N.b.: I found a 'system' audio control that allows web videos to play using PulseAudio !

**Reply by pcar75** [reply]
So far, 
after a few hours of tweaking/testing/rebooting, I found the desktop taskbar has a "Volume Control" (speaker) icon that I hadn't used much before. That little thing hides enough configuration behind, including a mixer.  It is in fact the PulseAudio (PulseAudioCtl hereafter) equivalent of QjackCtl, not the default system audio control.

Since Firefox does not have audio device configuration*, I use the PulseAudioCtl>Playback device to PulseAudioJackSink and reviewed the settings in the >OutputDevices and >Configuration tabs.
* I read somewhere that in Linux, most 'audio outputing' apps use PulseAudio (commercial apps) instead of Jack (studio apps).

VLC>Preferences>Audio>OutputModule is set to 'automatic' and I just change the VLC>Audio>AudioDevice to JackSink (PulseAudioJackSink) or 'Built-In Analog Stereo'.

Also it seems that "PCM2902 Audio Codec Analog Stereo" is the audio device (TI chip?) for my USB mixer and "Built-In Analog Stereo" is the default one for my pc audio card and speakers.
The 'GM107 High Definition Audio Controller [Geforce 940MX]' of my video card has not been configured and there is no need for it presently.

So I was finally able to have a Jamulus session with VLC and a YouTube video.
Thanks to all !
Now I got to figure out a way to automate all these procedures (scrip, lash or RaySession) ... mmmph, another day !
